docs/evm/truffle.md

```diff
@@ -0,0 +1,143 @@
+---
+id: truffle
+title: Using Truffle with the EVM
+sidebar_label: Using Truffle
+---
+
+[Truffle](https://trufflesuite.com) is a popular tool for EVM-based smart contract development and testing. A typical dApp using Truffle will have a file called `truffle-config.js` in the project root specifying details on connection, compilation, networks, and so on.
+
+NEAR has a custom provider called [`near-web3-provider`](https://github.com/near/near-web3-provider) that can be specified in truffle configuration. It's published as an [NPM package](https://npmjs.com/package/near-web3-provider) and can be added with:
+
+    npm install near-web3-provider --save-dev
+
+**Note**: you may also use `near-web3-provider` as the provider on the frontend, in which case the proper flag is `--save` instead of `--save-dev`.
+
+## Example `truffle-config.js` for NEAR
+
+```js
+const { NearProvider } = require('near-web3-provider');
+
+module.exports = {
+  networks: {
+    near_local: {
+      network_id: "*",
+      skipDryRun: true,
+      provider: () => new NearProvider({
+        networkId: 'local',
+        masterAccountId: 'test.near',
+      }),
+    },
+    near_betanet: {
+      network_id: "*",
+      skipDryRun: true,
+      provider: () => new NearProvider({
+        networkId: 'betanet',
+        masterAccountId: process.env.NEAR_MASTER_ACCOUNT
+      }),
+    },
+    near_testnet: {
+      network_id: "*",
+      skipDryRun: true,
+      provider: () => new NearProvider({
+        networkId: 'testnet',
+        masterAccountId: process.env.NEAR_MASTER_ACCOUNT
+      }),
+    },
+    near_mainnet: {
+      network_id: 1313161554, // See https://chainid.network
+      skipDryRun: true,
+      provider: () => new NearProvider({
+        networkId: 'mainnet',
+        masterAccountId: process.env.NEAR_MASTER_ACCOUNT
+      }),
+    },
+    develop: {
+      host: "127.0.0.1",
+      network_id: "*",
+      port: 8545
+    }
+  }
+};
+```
+
+To see an example of this in a project, see the [NEAR Pet Shop example](https://github.com/near-examples/near-pet-shop/blob/master/truffle-config.js).
+
+## Testing
+
+At this time, tests run using an existing account on the network. For example, if a user is going to test on betanet, they would first create a betanet account at:
+
+https://wallet.betanet.near.org
+
+Then use [NEAR CLI](/docs/development/near-cli) to login to that account, which creates a full-access key file in their home directory. (Specifically `~/.near-credentials`) Now tools and libraries like NEAR CLI and [`near-api-js`](https://github.com/near/near-api-js) can utilize these keys for actions including deploying contracts, transferring Ⓝ, creating subaccounts, etc. (See all available [Actions here](https://nomicon.io/RuntimeSpec/Actions.html).) The accounts created for testing are subaccounts (ex. `subaccount.yourname.betanet` created from `yourname.betanet`)
```

Comment:
"NEAR CLI" => "the NEAR CLI". (I get the feeling you're using this is as a proper name? That'd be clearer if more clearly referring to a project name, such as "near-cli".)

Reply:
"(Specifically `~/.near-credentials`)" => "(Specifically `~/.near-credentials`.)" (note the period)